src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -696,8 +696,9 @@ private void AddToBatch(CounterDetail counter)
                     ChangeVector = counter.ChangeVector
                 };
 
-                _cmd.Add(counter.DocumentId, counterOp);
-                _batchSize++;
+                _cmd.Add(counter.DocumentId, counterOp, out var isNew);
+                if (isNew)
+                    _batchSize++;
```

Comment:
does it mean that a batch will contain counters of 1024 different documents, with an unlimited amount of counters per doc ?

Reply:
No, each 1024 counters. But I think that we can safely increase the size back to 10,000, since this is now much cheaper, no?

Reply:
Are you sure? it looks like `isNew` is set to `true` only when the document id is new to this batch. Or am I missing something?

Reply:
To improve performance we are trying to aggregate as many counters for the same document as possible. And it is indeed unbounded.. should I limit it, or simply revert this change completely?

Reply:
Let's limit this to a high number, such as 32*1024 or something like this.